### ISSUE TITLE
Miri running forever

### ISSUE THREAD

**Issue by MaxCWhitehead** [fishfolk/bones#466]
Miri tests in CI seem to hang. Seems to have first started occurring on: e20862b2d, but unsure if related.

**Comment by MaxCWhitehead** [comment]
Ah it seems like `components::typed::tests::iter_with_bitset_optional` may be hanging. 

`cargo +nightly miri test` repros (can repro locally)

and these I cannot repro with:
`cargo +nightly test --workspace`
`cargo test --workspace` 

Not sure yet if issue with test or miri, but will look into it, and disable the test for now possibly.

**Comment by nelson137** [comment]
Sorry Max, just seeing this. I think many of the tests I've added are *extremely* slow on miri. The entities and query tests are a huge contributor to how long it takes the CI job to run because of the keysize -- miri hangs on a test when it has to iterate over the whole bitset. The default keysize if 16, meaning the bitsets are only 8 KB but apparently that's too much.

I've been trying to add some smarter checks in these iterators to make it stop sooner if it can detect that there's no reason to keep going, or even skip iteration entirely for empty bitsets (which are common in the tests), but getting the logic right is tricky. I should have a PR for this soon. I'm also experimenting with dropping the keysize to 10 (128 byte sets) which makes for dramatically faster test runs.

**With the optimizations I've made so far and using `keysize10` a single test run takes less than 2 minutes on my machine** (not including compilation time, and only 1 seed)! So I'm optimistically hoping that a full miri CI job run will take <40 min (that's including compilation time and the test runs for the 11 seeds).
